copt/constraint.py

```diff
@@ -20,52 +20,72 @@ def __call__(self, x):
         else:
             return np.infty
 
-    def prox(self, x, step_size):
+    def prox(self, x, step_size=None):
         return euclidean_proj_l1ball(x, self.alpha)
 
-    def lmo(self, u, x):
-        """Return s - x, s solving the linear problem
-    max_{||s||_1 <= alpha} <u, s>
+    def lmo(self, u, x, active_set=None):
+        """Linear Minimization Oracle.
+        
+        Return s - x with s solving the linear problem
+            max_{||s||_1 <= alpha} <u, s>
+        
+        Args:
+          u: usually -gradient
+          x: usually the iterate of the considered algorithm
+          active_set: no effect here.
+          
+        Returns:
+          update_direction: s - x, where s is the vertex of the constraint most correlated with u
+          fw_vertex_rep: a hashable representation of s, for active set management
```

Comment:
please add what type this would be (like for the others)

Reply:
Here it's (float, int); but it could be anything hashable as far as the algorithm is concerned.